core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -400,7 +477,7 @@ public boolean isEmpty() {
 
    @Override
    public boolean containsKey(Object key) {
-      return cacheImplementation.containsKey(key, flags);
+      return get(key) != null;
```

Comment:
I have to look again but the reason is because contains key required more changes in CacheImpl to get the lock owner to be set properly. I can spend the extra time to fix it though as well.